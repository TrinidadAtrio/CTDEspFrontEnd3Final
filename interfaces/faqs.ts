export interface Faqs {
    id: number;
    question: string;
    answer: string;
}

export type Data = Faqs[] | {message: string}