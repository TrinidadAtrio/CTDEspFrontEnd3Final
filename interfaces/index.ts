export * from './comics';
export * from './faqs';
export * from './character';