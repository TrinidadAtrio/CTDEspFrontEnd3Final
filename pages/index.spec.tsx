import {render, screen} from "@testing-library/react";
import IndexPage from "dh-marvel/pages/index.page";
import Index from "dh-marvel/pages/index.page";
import { Comics } from "interfaces";

describe('IndexPage', () => {
  const comics = {
    code: 123,
    status: 'ok',
    copyright: '',
    attributionText: '',
    attributionHTML: '',
    etag: '',
    data: {
      total: 100,
      limit: 5,
      results: [],
      offset: 10,
      count: 10,
    }
  } as Comics;
    describe('when rendering default', () => {
        it('should render empty home', () => {
            const comp = render(<Index comics={comics} page={1}/>)
            expect(comp.baseElement).toMatchSnapshot()
        })
    })

})