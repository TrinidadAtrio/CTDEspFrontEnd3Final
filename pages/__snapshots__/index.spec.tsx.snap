// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexPage when rendering default should render empty home 1`] = `
<body>
  <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
    >
      <div
        class="MuiStack-root css-fdaco8-MuiStack-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter MuiTypography-gutterBottom css-fqhh5w-MuiTypography-root"
        >
          Tus comics favoritos en un solo lugar
        </div>
        <div
          class="MuiStack-root css-v6ua4x-MuiStack-root"
        >
          <nav
            aria-label="pagination navigation"
            class="MuiPagination-root MuiPagination-text css-1oj2twp-MuiPagination-root"
          >
            <ul
              class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
            >
              <li>
                <button
                  aria-label="Go to previous page"
                  class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-disabled MuiPaginationItem-previousNext css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                    data-testid="NavigateBeforeIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                    />
                  </svg>
                </button>
              </li>
              <li>
                <button
                  aria-current="true"
                  aria-label="page 1"
                  class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-selected MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
                  tabindex="0"
                  type="button"
                >
                  1
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </li>
              <li>
                <button
                  aria-label="Go to page 2"
                  class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
                  tabindex="0"
                  type="button"
                >
                  2
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </li>
              <li>
                <button
                  aria-label="Go to page 3"
                  class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
                  tabindex="0"
                  type="button"
                >
                  3
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </li>
              <li>
                <button
                  aria-label="Go to page 4"
                  class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
                  tabindex="0"
                  type="button"
                >
                  4
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </li>
              <li>
                <button
                  aria-label="Go to page 5"
                  class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
                  tabindex="0"
                  type="button"
                >
                  5
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </li>
              <li>
                <div
                  class="MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-ellipsis css-1v2lvtn-MuiPaginationItem-root"
                >
                  â€¦
                </div>
              </li>
              <li>
                <button
                  aria-label="Go to page 9"
                  class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
                  tabindex="0"
                  type="button"
                >
                  9
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </li>
              <li>
                <button
                  aria-label="Go to next page"
                  class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-previousNext css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                    data-testid="NavigateNextIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </li>
            </ul>
          </nav>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-spacing-md-3 css-1xgjpwf-MuiGrid-root"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
